packages/swapper/src/swappers/ThorchainSwapper/endpoints.ts

```diff
@@ -328,13 +329,36 @@ export const thorchainApi: SwapperApi = {
 
   checkTradeStatus: async ({
     txHash,
+    chainId,
     config,
   }): Promise<{
     status: TxStatus
     buyTxHash: string | undefined
     message: string | undefined
   }> => {
     try {
+      const safeTransactionInfo = await fetchSafeTransactionInfo({
+        chainId,
+        maybeSafeTxHash: txHash,
+      })
+      const { isSafeTxHash, transaction } = safeTransactionInfo
+
+      // No buyTxHash handling is correct - we mutate with the actual on-chain transaction, meaning the regular flow then takes over
+      if (
+        isSafeTxHash &&
+        transaction?.confirmations &&
+        Number(transaction.confirmations.length) < transaction.confirmationsRequired
+      ) {
+        return {
+          status: TxStatus.Pending,
+          message: `SAFE proposal submitted. ${transaction.confirmations.length} out of ${transaction.confirmationsRequired} signed.`,
```

Comment:
translation here and is `SAFE proposal submitted` the correct vernacular here?

Reply:
Summoning @twblack88 - maybe *proposed* is a better vernacular?

Reply:
maybe queued? 
its basically this part right?

![Screenshot 2024-08-21 at 11 35 26 AM](https://github.com/user-attachments/assets/3faf1100-64d2-445f-8e5f-52823ceabf66)

Reply:
queued lines up. I was even talking about `propsal` like should it be `SAFE transaction queued.`?

Reply:
`SAFE transaction created.` even sounds fine to me.

Reply:
Precisely @twblack88 that's after the first signer signs i.e (assuming multiple signers, else disregard the first point, we effectively go straight to on-chain Tx by the time we poll

- First signer signs: Proposal is proposed/submitted/created. We have a SAFE hash, but no on-chain Tx here just yet
- Second signer signs: Assuming that's a 2 signers vault, then at this stage the proposal gets executed on-chain and we have a Txid (there may be a few smolish intermediary states but for all intents and purposes, we're not concerned about these)